lib/active_merchant/billing/gateways/fat_zebra.rb

```diff
@@ -122,11 +122,11 @@ def add_creditcard(post, creditcard, options = {})
       end
 
       def add_extra_options(post, options)
-        extra = {}
+        extra = options.fetch(:extra, {})
         extra[:ecm] = '32' if options[:recurring]
-        extra[:cavv] = options[:cavv] if options[:cavv]
-        extra[:xid] = options[:xid] if options[:xid]
-        extra[:sli] = options[:sli] if options[:sli]
+        extra[:cavv] = options[:cavv] || options.dig(:three_d_secure, :cavv) if options[:cavv] || options.dig(:three_d_secure, :cavv)
+        extra[:xid] = options[:xid] || options.dig(:three_d_secure, :xid) if options[:xid] || options.dig(:three_d_secure, :xid)
+        extra[:sli] = options[:sli] || options.dig(:three_d_secure, :eci) if options[:sli] || options.dig(:three_d_secure, :eci)
```

Comment:
FZ acknowledge they have used the Visa specific term here but the value is the same.